src/Education/Rule/SLCoreRuleMetaDataProvider.cs

```diff
@@ -74,7 +72,35 @@ public async Task<IRuleInfo> GetRuleInfoAsync(SonarCompositeRuleId ruleId)
         return null;
     }
 
-    private static RuleInfo Convert(EffectiveRuleDetailsDto effectiveRuleDetailsAsync) =>
+    /// <inheritdoc />
+    public async Task<IRuleInfo> GetRuleInfoAsync(SonarCompositeRuleId ruleId, Guid? issueId)
```

Comment:
This looks like an extension method.

Reply:
The GetRuleInfoAsync()? Is can not an extension method. It's part of the service's contract

Reply:
As discussed, I only exposed one method in the IRuleMetadataProvider